Stripe/PublicHeaders/STPSourceParams.h

```diff
@@ -0,0 +1,200 @@
+//
+//  STPSourceParams.h
+//  Stripe
+//
+//  Created by Ben Guo on 1/23/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "STPFormEncodable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class STPCardParams;
+
+/**
+ *  An object representing parameters used to create a Source object. 
+ *  @see https://stripe.com/docs/api#create_source
+ */
+@interface STPSourceParams : NSObject<STPFormEncodable, NSCopying>
+
+/**
+ *  The type of the source to create. Required.
+ */
+@property (nonatomic, copy, nullable) NSString *type;
```

Comment:
Yeah, I thought about this a bit more, and I think it's a reasonable tradeoff to make – I don't think it'll be necessary to add new source types so often that it's worth abandoning the type safety.